src/main/scala/clump/Clump.scala

```diff
@@ -62,11 +62,15 @@ object Clump {
 
   def collect[T](clumps: List[Clump[T]]): Clump[List[T]] = new ClumpCollect(clumps)
 
-  def sourceFrom[T, U, C](fetch: C => Future[Map[T, U]])(implicit cbf: CanBuildFrom[Nothing, T, C]): ClumpSource[T, U] =
-    ClumpSource.from(fetch)
-
-  def source[T, U, C](fetch: C => Future[Iterable[U]])(keyExtractor: U => T)(implicit cbf: CanBuildFrom[Nothing, T, C]): ClumpSource[T, U] =
-    ClumpSource(fetch)(keyExtractor)
+  def source[C] = new {
+    def apply[T, U](fetch: C => Future[Iterable[U]])(keyExtractor: U => T)(implicit cbf: CanBuildFrom[Nothing, T, C]): ClumpSource[T, U] =
```

Comment:
I was just wondering, rather than explicitly providing a type `C` here, can't we just do
```scala
(implicit cbf: CanBuildFrom[Nothing, T, Iterable[T])
```

I guess it won't work because we need the type of Iterable